frontend/tools/translationKey/generateTranslationKeyType.ts

```diff
@@ -0,0 +1,52 @@
+import fs from 'fs';
+import path from 'path';
+
+/**
+ * Recursive function which returns sorted array of flattened keys.
+ *
+ * For example `{a: 'abc', b: {c: 'def'}}` is returned as `['<prefix>.a', '<prefix>.b.c']`
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const flattenKeys = (obj: any, prefix: string): Array<string> => {
```

Comment:
Sorry, I didn't mention this before, but let's use the `function flattenKeys(...)` format for functions instead of `const ...`, because it makes the code more legible and to my knowledge there are no real benefits anymore of using the latter format.

This way we can also move the functions to the bottom of the file for added legibility.